### PR TITLE
Adding a top-level store for the checkout iframe

### PR DESCRIPTION
**Pull request by cnasc**
# Description

This PR introduces a top-level store for the checkout iframe. I'm keeping this separate from the existing Redux store because there shouldn't be any cross-pollination between them. The basic step was to find all instances of `useState` and replace them with updates dispatched to the store. This lays the foundation for easier switching between views without dumping all our state (important for metadata).

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [x] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by cnasc** [comment]
This needs some additional tests before it passes, which I am working on now

File: unlock-app/src/pages/_app.js [hunk 1]
```diff
@@ -155,9 +156,11 @@ The Unlock team
                 <Web3ServiceProvider value={web3Service}>
                   <WalletServiceProvider value={walletService}>
                     <ConfigProvider value={config}>
-                      <GlobalErrorConsumer>
-                        <Component {...pageProps} />
-                      </GlobalErrorConsumer>
+                      <CheckoutStoreProvider>
```
Comment (julien51) [inline unlock-app/src/pages/_app.js]: Would it make more sense to put that one level down, inside of a `FullCheckoutWrapper` component?

Comment (julien51) [reply unlock-app/src/pages/_app.js]: or maybe actually the existing `CheckoutContainer` ?

Comment (julien51) [reply unlock-app/src/pages/_app.js]: so that we keep the `App` stuff only for things which are "global"?

**Comment by julien51** [comment]
I think this is generally good, but I have a tension around the use of `_app`. Could we wrap stuff inside of `CheckoutStoreProvider` at a "lower" level, like `CheckoutContainer`?

**Comment by julien51** [comment]
This was it does not "pollute" the rest of the application?

**Comment by cnasc** [comment]
Moved down a level from `_app`, ready for another review

**Pull request merged**